### PR TITLE
Do not display metric/param names that only belong to archived runs 

### PR DESCRIPTION
**Pull request by dsuhinin**

### CODE REVIEW

File: pkg/api/aim/projects.go [hunk 1]
```diff
@@ -131,7 +131,15 @@ func GetProjectParams(c *fiber.Ctx) error {
 			resp[s] = fiber.Map{}
 		case "metric":
 			var metricKeys []string
-			if tx := database.DB.Model(&database.LatestMetric{}).Distinct().Pluck("Key", &metricKeys); tx.Error != nil {
+			if tx := database.DB.Distinct().Model(
+				&database.LatestMetric{},
+			).Joins(
+				"JOIN runs USING(run_uuid)",
+			).Where(
+				"runs.lifecycle_stage <> ?", database.LifecycleStageDeleted,
```
Comment (jgiannuzzi) [inline pkg/api/aim/projects.go:139]: for consistency with the rest of the codebase, let's instead check that the lifecycle stage is active
```suggestion
				"runs.lifecycle_stage = ?", database.LifecycleStageActive,
```

File: pkg/api/aim/projects.go
Comment (jgiannuzzi) [inline pkg/api/aim/projects.go:1]: the same logic needs to be applied to param and tag names!

File: pkg/api/aim/api/response/project.go [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+				ExampleType string `json:"__example_type__"`
+			} `json:"mlflow.source.name"`
+			MlflowSourceType struct {
+				ExampleType string `json:"__example_type__"`
+			} `json:"mlflow.source.type"`
+			MlflowUser struct {
+				ExampleType string `json:"__example_type__"`
+			} `json:"mlflow.user"`
+		} `json:"tags"`
+	} `json:"params"`
```
Comment (jgiannuzzi) [inline pkg/api/aim/api/response/project.go:18]: params are an arbitrary `map[string]any`!

File: tests/integration/golang/aim/project/get_project_params_test.go [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+	// 3. check that response contains metric from previous step.
+	resp := response.ProjectParamsResponse{}
+	err = s.client.DoGetRequest(
+		"/projects/params?sequence=metric",
+		&resp,
+	)
+	assert.Nil(s.T(), err)
+	assert.Equal(s.T(), 1, len(resp.Metric))
+	_, ok := resp.Metric[metric.Key]
+	assert.True(s.T(), ok)
```
Comment (jgiannuzzi) [inline tests/integration/golang/aim/project/get_project_params_test.go:90]: please check this with params and tags as well

**Review by jgiannuzzi — CHANGES_REQUESTED** [review]

File: pkg/api/aim/projects.go (hunk 1 above)
Comment (dsuhinin) [reply pkg/api/aim/projects.go]: done.

File: pkg/api/aim/api/response/project.go (hunk 2 above)
Comment (dsuhinin) [reply pkg/api/aim/api/response/project.go]: done.

File: tests/integration/golang/aim/project/get_project_params_test.go (hunk 3 above)
Comment (dsuhinin) [reply tests/integration/golang/aim/project/get_project_params_test.go:115]: checked.

File: pkg/api/aim/projects.go [hunk 4]
```diff
@@ -131,7 +147,15 @@ func GetProjectParams(c *fiber.Ctx) error {
 			resp[s] = fiber.Map{}
 		case "metric":
 			var metricKeys []string
-			if tx := database.DB.Model(&database.LatestMetric{}).Distinct().Pluck("Key", &metricKeys); tx.Error != nil {
+			if tx := database.DB.Distinct().Model(
```
Comment (jgiannuzzi) [inline pkg/api/aim/projects.go:150]: nitpick: please move the call to `Distinct()` to a similar place as for params and tags

**Review by jgiannuzzi — COMMENTED** [review]
apart from the nitpick below, LGTM 👍

File: pkg/api/aim/projects.go (hunk 4 above)
Comment (dsuhinin) [reply pkg/api/aim/projects.go:150]: sure.

Comment (dsuhinin) [reply pkg/api/aim/projects.go:150]: done!

**Review by jgiannuzzi — APPROVED** [review]

**Pull request merged**